File: Programmers/같은 숫자는 싫어.cpp
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> arr)
{
	vector<int> answer;

	for (int i = 0; i < arr.size(); i++)
	{
		if (!answer.empty() && answer.back() == arr[i])
			continue;
		answer.push_back(arr[i]);
	}
	return answer;
}
int main() {
	vector<int> arr = { 1, 1, 3, 3, 0, 1, 1 };
	vector<int> test = solution(arr);
	for (int i = 0; i < test.size(); i++)
	{
		cout << test[i];
	}
	system("pause");
}

File: Programmers/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        ArrayList<Integer> al = new ArrayList<>();
        for(int i = 0; i < arr.length; i++)
            if(!al.isEmpty() && al.get(al.size()-1) == arr[i])
                continue;
            else
                al.add(arr[i]);
        int[] answer = new int[al.size()]; 
        for(int i = 0; i < al.size(); i++)
            answer[i] = al.get(i).intValue();
        return answer;
    }
}
